src/workers/postUpdated.ts

```diff
@@ -254,11 +263,20 @@ const updatePost = async ({
   questions,
   content_type = PostType.Article,
 }: UpdatePostProps) => {
-  const postType = contentTypeFromPostType[content_type];
-  const databasePost = await entityManager
+  let postType = contentTypeFromPostType[content_type];
+  let databasePost = await entityManager
     .getRepository(postType)
     .findOneBy({ id });
 
+  // If we don't find the post, we need to check if it's a youtube video and
+  // try to find it again as an article
+  if (!databasePost && content_type === PostType.VideoYouTube) {
+    postType = ArticlePost;
+    databasePost = await entityManager
+      .getRepository(postType)
+      .findOneBy({ id });
+  }
```

Comment:
I don't particularly like this part, but not sure if there's any better methods to re-fetch it with a different repository selected.

Reply:
This indeed is limitation on typeorm side. Maybe we can do a generic sql query for just the type in the db and then do a proper query when we know what is in the DB?

Reply:
Can't we do something like `getRepository(Post)` and see if the type matches or not?

I think I'm also missing the actual part where you change the type right? (save it as such type)

Reply:
The change of type happens in [allowedFieldsMapping](https://github.com/dailydotdev/daily-api/blob/db94e7a148551f90fdb1eb4db9bd0b8f5c65f8e6/src/workers/postUpdated.ts). I added `type` (and others) [to the mapping](https://github.com/dailydotdev/daily-api/blob/db94e7a148551f90fdb1eb4db9bd0b8f5c65f8e6/src/workers/postUpdated.ts#L219).

Reply:
So I think it might need to be a bit more advanced than what I wrote here.
Since `ArticlePost` doesn't have `videoId`, updating it fails.